python/ray/train/_internal/session.py

```diff
@@ -478,7 +483,8 @@ def get_dataset_shard(
             warnings.warn(
                 "No dataset passed in. Returning None. Make sure to "
                 "pass in a Dataset to Trainer.run to use this "
-                "function."
+                "function.",
+                stacklevel=2,
```

Comment:
nit, define a constant?

Reply:
Sure. I'll do a full pass of all warning calls in a separate PR. This is not quite relevant to `TrainStateActor` here.